make check segfaults on OS X

Current master (797279c7143bd2a379f56195112c3bfba2f8302d) segfaults on `make check` on OS X.


```
$ uname -a
Darwin nb-skaes2.local 13.4.0 Darwin Kernel Version 13.4.0: Wed Mar 18 16:20:14 PDT 2015; root:xnu-2422.115.14~1/RELEASE_X86_64 x86_64
$ make check
Making check in doc
make[1]: Nothing to be done for `check'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make  src/czmq_selftest
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS check-local
============================================================================
Testsuite summary for czmq 3.0.1
============================================================================
# TOTAL: 0
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
/bin/sh ./libtool --mode=execute ./src/czmq_selftest
Running czmq selftests...
 * zactor: OK
 * zauth: OK
 * zarmour: OK
 * zbeacon: OK
 * zcert: OK
 * zcertstore: OK
 * zchunk: OK
 * zclock: OK
 * zconfig: OK
 * zdigest: OK
 * zdir: OK
 * zdir_patch: OK
 * zfile: OK
 * zframe: OK
 * zgossip: OK
 * zhashx: OK
 * ziflist: OK
 * zlistx: OK
 * zloop: OK
 * zmonitor: OK
 * zmsg: OK
 * zpoller: OK
 * zproxy: OK
 * zrex: OK
make[2]: *** [check-local] Segmentation fault: 11
make[1]: *** [check-am] Error 2
make: *** [check-recursive] Error 1
`
```

--- a/src/zsock.c
+++ b/src/zsock.c
@@ -1652,7 +1652,7 @@ zsock_test (bool verbose)
     char *original = "pointer";
 
     //  Test zsock_recv into each supported type
-    zsock_send (writer, "i1248zsbcfuhp",
+    zsock_send (writer, "i1248zsbcfUhp",
                 -12345, number1, number2, number4, number8,
                 "This is a string", "ABCDE", 5,
                 chunk, frame, uuid, hash, original);
@@ -1667,7 +1667,7 @@ zsock_test (bool verbose)
     size_t size;
     char *pointer;
     number8 = number4 = number2 = number1 = 0;
-    rc = zsock_recv (reader, "i1248zsbcfuhp",
+    rc = zsock_recv (reader, "i1248zsbcfUhp",
                      &integer, &number1, &number2, &number4, &number8,
                      &string, &data, &size, &chunk, &frame, &uuid, &hash, &pointer);
     assert (rc == 0);

